File: src/main/java/com/senderman/lastkatkabot/command/user/Divorce.java
package com.senderman.lastkatkabot.command.user;

import com.annimon.tgbotsmodule.api.methods.Methods;
import com.annimon.tgbotsmodule.services.CommonAbsSender;
import com.senderman.lastkatkabot.ApiRequests;
import com.senderman.lastkatkabot.command.CommandExecutor;
import com.senderman.lastkatkabot.dbservice.UserStatsService;
import org.springframework.stereotype.Component;
import org.telegram.telegrambots.meta.api.objects.Message;

@Component
public class Divorce implements CommandExecutor {

    private final UserStatsService users;

    public Divorce(UserStatsService users) {
        this.users = users;
    }

    @Override
    public String getTrigger() {
        return "/divorce";
    }

    @Override
    public String getDescription() {
        return "подать на развод";
    }

    @Override
    public void execute(Message message, CommonAbsSender telegram) {

        var userId = message.getFrom().getId();
        var userStats = users.findById(userId);
        var loverId = userStats.getLoverId();

        if (loverId == null) {
            ApiRequests.answerMessage(message, "У вас и так никого нет!").callAsync(telegram);
            return;
        }

        var loverStats = users.findById(loverId);

        userStats.setLoverId(null);
        loverStats.setLoverId(null);
        users.save(userStats);
        users.save(loverStats);

        ApiRequests.answerMessage(message, "Вы расстались со своей половинкой!").callAsync(telegram);
        Methods.sendMessage(loverId, "Ваша половинка решила с вами расстаться :(").callAsync(telegram);

    }
}

File: src/main/kotlin/com/senderman/lastkatkabot/usercommands/Divorce.kt
package com.senderman.lastkatkabot.usercommands

import com.annimon.tgbotsmodule.api.methods.Methods
import com.senderman.lastkatkabot.LastkatkaBotHandler
import com.senderman.lastkatkabot.Services
import com.senderman.neblib.CommandExecutor
import com.senderman.neblib.TgUser
import org.telegram.telegrambots.meta.api.methods.send.SendMessage
import org.telegram.telegrambots.meta.api.objects.Message

class Divorce(private val handler: LastkatkaBotHandler) : CommandExecutor {
    override val command: String
        get() = "/divorce"
    override val desc: String
        get() = "подать на развод"

    override fun execute(message: Message) {
        val chatId = message.chatId
        val userId = message.from.id
        val loverId = Services.db.getLover(userId)
        if (loverId == 0) {
            handler.sendMessage(chatId, "У вас и так никого нет!")
            return
        }
        Services.db.divorce(userId)
        handler.sendMessage(chatId, "Вы расстались со своей половинкой! А ведь так все хорошо начиналось...")
        val user = TgUser(Methods.getChatMember(userId.toLong(), userId).call(handler).user)
        try {
            handler.execute(
                SendMessage(
                    loverId.toLong(),
                    "Ваша половинка (${user.link}) покинула вас... Теперь вы одни..."
                ).enableHtml(true)
            )
        } catch (ignored: Exception) {
        }
    }
}